src/components/Header/NotificationList.tsx

```diff
@@ -1,40 +1,33 @@
 import React from 'react'
 import { useTranslation } from 'react-i18next'
 import { NotificationIcon } from '@components/Icons/NotificationIcon'
+import { NotificationsLiatIcon } from '@components/Icons/NotificationsLiatIcon'
 import { Translation } from '@constants/translations'
 import { Menu } from '@headlessui/react'
-import Image from 'next/image'
+import { dispatch, useAppSelector } from '@redux/hooks'
+import { applicationsMiddleware, applicationsSelector } from '@redux/slices/applications'
+import { Button } from '@uiComponents/Button'
+import { useRouter } from 'next/router'
 
-const notificationsList = [
-  {
-    id: 1,
-    image: '/image1.jpg',
-    paragraph: 'Lorem ipsum dolor sit amet consectetur. Posuere diam.',
-  },
-  {
-    id: 2,
-    image: '/image1.jpg',
-    paragraph: 'Lorem ipsum dolor sit amet consectetur. Posuere diam.',
-  },
-  {
-    id: 3,
-    image: '/image1.jpg',
-    paragraph: 'Lorem ipsum dolor sit amet consectetur. Posuere diam.',
-  },
-  {
-    id: 4,
-    image: '/image1.jpg',
-    paragraph: 'Lorem ipsum dolor sit amet consectetur. Posuere diam.',
-  },
-  {
-    id: 5,
-    image: '/image1.jpg',
-    paragraph: 'Lorem ipsum dolor sit amet consectetur. Posuere diam.',
-  },
-]
+interface ChildComponentProps {
+  close?: () => void
+}
 
-export const NotificationList = () => {
+export const NotificationList = ({ close }: ChildComponentProps) => {
   const [t] = useTranslation()
+  const router = useRouter()
+
+  const { notifications } = useAppSelector(applicationsSelector.applications)
+
+  const handleClick = (id: string) => {
+    dispatch(applicationsMiddleware.getNotificationsId(id))
+
+    router.push('/profile/applications')
+
+    if (typeof close === 'function') {
+      close()
+    }
```

Comment:
this code is not reachable and what is this condition?

Reply:
i raise it before routing